src/Entity/Form/TprDeleteForm.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\helfi_tpr\Entity\Form;
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Entity\ContentEntityDeleteForm;
+use Drupal\Core\Entity\EntityRepositoryInterface;
+use Drupal\Core\Entity\EntityTypeBundleInfoInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Utility\Error;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\ClientException;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Tpr-entity delete form.
+ */
+final class TprDeleteForm extends ContentEntityDeleteForm {
+
+  /**
+   * The constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityRepositoryInterface $entity_repository
+   *   The entity repository.
+   * @param \Drupal\Core\Entity\EntityTypeBundleInfoInterface $entity_type_bundle_info
+   *   The entity type bundle info interface.
+   * @param \Drupal\Component\Datetime\TimeInterface $time
+   *   The time interface.
+   * @param \GuzzleHttp\ClientInterface $http_client
+   *   The time interface.
+   */
+  public function __construct(
+    EntityRepositoryInterface $entity_repository,
+    EntityTypeBundleInfoInterface $entity_type_bundle_info,
+    TimeInterface $time,
+    private readonly ClientInterface $http_client,
+  ) {
+    parent::__construct($entity_repository, $entity_type_bundle_info, $time);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity.repository'),
+      $container->get('entity_type.bundle.info'),
+      $container->get('datetime.time'),
+      $container->get('http_client'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    if ($this->helfiTprEntityExists()) {
+      $this->messenger()->addWarning(
+        $this->t('Cannot delete TPR-entity which still exists in the API')
+      );
+      return;
+    }
+
+    parent::submitForm($form, $form_state);
+  }
+
+  /**
+   * Check if the entity being deleted exists in the API.
+   *
+   * @return bool
+   *   The entity exists in api.
+   */
+  private function helfiTprEntityExists(): bool {
```

Comment:
This can be renamed to entityExists since the helfi + tpr context is already known.